### ISSUE TITLE
dropNode error: e is undefined  (sigma.min.js:2:45789)

### ISSUE THREAD

**Issue by gitbestsou** [jacomyal/sigma.js#954]
$("#deletenode").click(function(){
if(deleteflag===0)
	{
		alert("No Current Node");
	}
else
	{
		console.log(nodeToBeDeleted);
		s.graph.dropNode(nodeToBeDeleted);
		s.refresh();
	}
});

Whenever the function is being executed and the deleteflag is not 0, it is showing the error in the issue title. What is the problem? s is globally declared. nodeToBeDeleted is the id of the node to be deleted which I am getting from some other method.

**Comment by Yomguithereal** [comment]
I cannot say with only this code. Does your console.log display an actual node id?

**Comment by gitbestsou** [comment]
Yes. No error there. It displays a valid node id. What can be the possible problems?

**Comment by Yomguithereal** [comment]
After further investigation it seems to mean there is an issue with the edge quad tree. The person that coded it seems to forgot to take into account for edge deletion. Do you need edge hovering in your use case?

**Comment by gitbestsou** [comment]
I am not sure what you referred by edge quad tree. I do not want to delete edge. I want to delete nodes when a particular button(having id 'deletenode') is clicked. nodeToBeDeleted is showing a valid node id. So,what is the problem in node deletion? What is the relevance of edge hovering here? 

Is there any other way to delete a node?

**Comment by gitbestsou** [comment]
Yes, I know that. I do not need edge hovering. What is the problem?

**Comment by Yomguithereal** [comment]
The problem seems to be that you enabled edge hovering and this has bugs related to edge deletion. But I may be mistakeb’n. Without more pieces of your code or a way to reproduce on my end it is very hard to tell.

**Comment by gitbestsou** [comment]
s = new sigma({graph: mygraph});
//mygraph is the graph object containing nodes and edges.
$("#deletenode").click(function(){
console.log(nodes.length);
if(deleteflag===0)
	{
		alert("No Current Node");
	}
else
	{
		console.log(nodeToBeDeleted);
		s.graph.nodes().forEach(
		function(n)
		{
    		if(n.id===nodeToBeDeleted)
				{
					s.graph.dropNode(n.id);
				}
		});
	}
console.log(nodes.length);
s.refresh();
});
 I have changed a bit. Still the same error.

var settings_config = {
						minNodeSize:4,
			 	 		maxNodeSize:5,
						minEdgeSize:1,
						maxEdgeSize:1.5,
						animationsTime:1,
						labelThreshold:20,
						enableEdgeHovering:true,
						enableHovering:false
					};

My settings is set to this now. If you are telling me to set enableEdgeHovering to false,that would make  clickEdge event stop to function,which I have used and very necessary in what I am developing. I cannot change that. So ,is the conclusion - I cannot add clickEdge and dropNode in the same program?
